### ISSUE TITLE
Why can't I use type-safe processAot in gradle build-logic

### ISSUE THREAD

**Issue by JoyfulAndSpeedyMan** [spring-projects/spring-boot#38200]
**Describe the problem**
I'm using gradle to build my project, I've defined a plugin 'spring-boot-native-dubbo-application', I want to configure some native public configuration in it, But I found it impossible to type safely use 'tasks.processAot' in the plugin 'spring-boot-native-dubbo-application' :
![image](https://github.com/graalvm/native-build-tools/assets/51226051/da55847b-e48b-49b8-9e17-14986badb6a6)
![image](https://github.com/graalvm/native-build-tools/assets/51226051/2e043f64-a5c5-4018-80a3-c379e515341f)
But in a project that uses the 'spring-boot-native-dubbo-application' plugin, it is possible to type safely configure 'tasks.processAot'
![image](https://github.com/graalvm/native-build-tools/assets/51226051/cfb3e035-6f5c-45f3-93ef-f0127ebd6e5f)
![image](https://github.com/graalvm/native-build-tools/assets/51226051/3b398c95-f230-4995-a6d8-67df9214fe8f)
But that would be tedious, because it would have to be configured for every project that uses native. Is there any way to configure 'tasks.processAot' in a common way?

**System Info (please complete the following information):**
 - OS: window10
 - GraalVM Version 8.4
 - Java Version graalvm 21
 - Plugin version 0.9.28

**Comment by wilkinsona** [comment]
Sorry, I don't understand the problem. It appears to be an issue with Gradle Kotlin scripts but I'm not sure how Spring Boot's Gradle plugin can have caused that. Perhaps you can provide a minimal sample that reproduces the problem? The screenshots aren't very informative I'm afraid.

**Comment by wilkinsona** [comment]
It looks like a bug in your IDE to me. Everything works on the command line.

**Comment by wilkinsona** [comment]
Thanks. I see the error now. It still doesn't look like a Spring Boot problem though. It appears to be a problem with type interference at compile time and Spring Boot has no influence over that. I suspect it's because your `com.example.spring-boot-application` has declared `org.springframework.boot:org.springframework.boot.gradle.plugin` as an `implementation` dependency so it's only visible to consumers at runtime and not at compile time.

**Comment by JoyfulAndSpeedyMan** [comment]
Yes, if the declaration `org.springframework.boot:org.springframework.boot.gradle.plugin` is displayed, this problem will not occur, should this question be asked to the gradle repository?

**Comment by wilkinsona** [comment]
No, I don't think so. I think you need to correct your `com.example.spring-boot-application` so that it declares `org.springframework.boot:org.springframework.boot.gradle.plugin` as an `api` dependency. This will make it visible to consumers at compile time and fix the compilation problem in your `com.example.spring-boot-native-application` plugin.

**Comment by JoyfulAndSpeedyMan** [comment]
Thank you very much, using api() way can normally pass `org.springframework.boot:org.springframework.boot.gradle.plugin` plugin dependencies, the problem can be solved perfectly.
